### PR TITLE
Created documentation page for code contributions.

### PR DESCRIPTION
**Pull request by jpakkane**
These are the things I came up with. Anything else people want to add for this before merging?

### CODE REVIEW

File: contributing.txt [hunk 1]
```diff
@@ -1,70 +1,4 @@
...
-Merge requests adding either of these two features will be automatically
-rejected. Please save everyone's time (especially your own) and don't start
-working on these features.
-
-
-Do I need to sign a CLA?
-
-No. All contributions are welcome.
+Contribution instructions can be found [on the
+website](http://mesonbuild.com/Contributing.html).
```
Comment (nirbheek) [inline contributing.txt]: The URL gives a 404, and this PR doesn't add it?

Comment (jpakkane) [reply contributing.txt]: It should. See the modification on `sitemap.txt` below.

Comment (nirbheek) [reply contributing.txt]: Am I blind or does `Contributing.md` not exist already and it's not added by this PR?

Comment (jpakkane) [reply contributing.txt]: :facepalm: Missing `git add`. Should be fixed now.

**Pull request merged**